dhis2/dhis2-core
# fix: viewing job progress errors requires special authority [DHIS2-16500]

### Summary
Requires a new authority `F_SCHEDULING_ANALYSE` to view progress errors.
This affects all endpoints that do directly or indirectly expose the errors.

For endpoints that only indirectly expose the errors as part of the overall progress structure the errors are cleared for users lacking authority whereas users having the authority can see them.

FYI: There are more endpoints returning the job progress information in `SchedulingController` but these are extracting only the tracking sequence part so the errors are not included there.

### Automatic Testing
Added new test scenarios that check the endpoints give a FORBIDDEN response for users with lacking authority for the negative cases.

The existing scenarios that check the different error endpoints for content now run with a special user that has the new `F_SCHEDULING_ANALYSE` authority and thereby test the positive cases.

### Manual Testing
* run a metadata import with a file that has error, e.g. uses a UID that does not exist
* find the UID of the import job by going to `/jobConfigurations/errors`
* login as a user with no special authority
* check `/jobConfigurations/errors`  is `FORBIDDEN`
* check `/jobConfigurations/{uid}/errors` is `FORBIDDEN`
* check `/jobConfigurations/{uid}/progress/errors`  is `FORBIDDEN`
* check `/jobConfigurations/{uid}/progress` is `FORBIDDEN`
* login as a user with the `F_SCHEDULING_ANALYSE` authority
* check that the above URLs give you access

Note that for endpoints that target a specific job the user that is configured as the executing user may also access the information. For a usual job this would be left empty (admin) so no additional user can access the information.
